src/sqlfluff/core/templaters/python.py

```diff
@@ -818,15 +818,15 @@ def _split_uniques_coalesce_rest(
                     # all as a tuple, because if we could do any better
                     # we would have done it by now.
 
-                    source_slice: Optional[slice] = None
+                    # Can we identify a meaningful portion of the patch
+                    # to recurse a split?
+                    sub_section: Optional[List[RawFileSlice]] = None
                     # If it's the start, the slicing is easy
                     if starts[1] == int_file_slice.templated_slice.stop:
-                        source_slice = slice(starts[0], raw_idx)
                         sub_section = int_file_slice.slice_buffer[:this_owu_idx]
                     # If we are AFTER the previous in the template, then it's
                     # also easy. [assuming it's not the same owu]
                     elif raw_idx > starts[0] and last_owu_idx != this_owu_idx:
-                        source_slice = slice(starts[0], raw_idx)
                         if last_owu_idx:
```

Comment:
@barrywhart I think that this version of the logic is much cleaner. thoughts?

Reply:
I agree -- if one variable is totally dependent on another, eliminating it or narrowing the scope within which it exists is 💯.